go/enclave/l2chain/l2_chain.go

```diff
@@ -323,15 +323,24 @@ func (lc *L2Chain) insertBlockIntoL1Chain(block *types.Block, isLatest bool) (*b
 		}
 
 		// ingested block is on a different branch to the previously ingested block - we may have to rewind L2 state
-		return &blockIngestionType{latest: isLatest, fork: true, preGenesis: false}, nil
+		return &blockIngestionType{isLatest: isLatest, fork: true, preGenesis: false}, nil
 	}
 
 	// this is the typical, happy-path case. The ingested block's parent was the previously ingested block.
-	return &blockIngestionType{latest: isLatest, fork: false, preGenesis: false}, nil
+	return &blockIngestionType{isLatest: isLatest, fork: false, preGenesis: false}, nil
 }
 
 // Updates the L1 and L2 chain heads, and returns the new L2 head hash and the produced batch, if there is one.
-func (lc *L2Chain) updateL1AndL2Heads(block *types.Block, isLatestBlock bool) (*common.L2RootHash, *core.Batch, error) {
+func (lc *L2Chain) updateL1AndL2Heads(block *types.Block, ingestionType *blockIngestionType) (*common.L2RootHash, *core.Batch, error) {
```

Comment:
while reading this function, I realised that "lc" is probably not the best variable name for the "layer2chain".
Maybe oc - from ObscuroChain?